crates/matrix-sdk-ui/src/lib.rs

```diff
@@ -37,3 +37,6 @@ fn init_logging() {
         .with(tracing_subscriber::fmt::layer().with_test_writer())
         .init();
 }
+
+#[cfg(feature = "experimental-uniffi")]
+uniffi::setup_scaffolding!();
```

Comment:
This approach will make UniFFI spreads like a virus in all our stack. I don't know what to think about this :-(.

Reply:
In a way, yes. But don't you think it's better to have some `cfg`-conditional pieces of code in the regular Rust crates than to repeat all the type definitions somewhere in the FFI crate?

Reply:
That's exactly what I don't know. It's annoying to have those definition duplications, and `From<A> for A'` in our code, I confess, but at least, changing something in one crate won't result in a breaking change in the FFI crate, and thus in the bindings. That's a “safe barrier”. In practice it's unlikely to happen (since there is the `cfg`, we _know_ it might break something), but eh, just thinking out loud.